pkg/codegen/docs/templates/function.tmpl

```diff
@@ -0,0 +1,59 @@
+{{ template "header" .Header }}
+
+{{ htmlSafe .Comment }}
+
+{{ .DeprecationMessage }}
+
+<!-- Input properties -->
+## Using {{ .Header.Title }}
+
+{{ htmlSafe "{{< langchoose csharp nojavascript >}}" }}
+
+<!-- TS/JS -->
+<div class="highlight"><pre class="chroma"><code class="language-typescript" data-lang="typescript"><span class="k">function </span>get{{ .ResourceName }}<span class="p">(</span>{{ htmlSafe .FunctionArgs.nodejs }}<span class="p">): Promise<{{ template "linkify_param" .FunctionResult.nodejs }}></span></code></pre></div>
+
+<!-- Python -->
+<div class="highlight"><pre class="chroma"><code class="language-python" data-lang="python"><span class="k">function </span> get_{{ pyName .ResourceName }}(</span>{{ htmlSafe .FunctionArgs.python }}<span class="p">)</span></code></pre></div>
+
+<!-- Go -->
+<div class="highlight"><pre class="chroma"><code class="language-go" data-lang="go"><span class="k">func </span>Lookup{{ .ResourceName }}<span class="p">(</span>{{ htmlSafe .FunctionArgs.go }}<span class="p">) (*{{ template "linkify_param" .FunctionResult.go }}, error)</span></code></pre></div>
+
+<!-- C# -->
+<div class="highlight"><pre class="chroma"><code class="language-csharp" data-lang="csharp"><span class="k">public static </span>Task<{{ template "linkify_param" .FunctionResult.csharp }}> <span class="p">Get{{ .ResourceName }}(</span>{{ htmlSafe .FunctionArgs.csharp }}<span class="p">)</span></code></pre></div>
+
+{{ if ne (len .InputProperties) 0 }}
+
+The following arguments are supported:
+{{ template "properties" .InputProperties }}
+
+{{ end }}
+
+<!-- Output properties -->
+## {{.Header.Title}} Output Properties
```

Comment:
Instead of "Output Properties" terminology, should we just use the name of the result type, and display it similar to the other Supporting Types? Or otherwise say something like `{{ .Header.Title }} Result`? The reason is, this is a bit different than output properties of resources. In that case, it's properties on the resource type itself. Whereas here, we're talking about properties on a result struct/class/interface.

Reply:
> Or otherwise say something like `{{ .Header.Title }} Result`?

Great point! And I like this suggestion.